src/Npgsql/NpgsqlCommand.cs

```diff
@@ -456,8 +456,13 @@ void DeriveParametersForFunction()
                 {
                     var param = new NpgsqlParameter();
 
-                    // TODO: Fix enums, composite types
-                    param.NpgsqlDbType = c.Connection.Connector.TypeMapper.GetNpgsqlTypeByOid(types[i]);
+                    (var npgsqlDbType, var postgresType, var dataTypeName) =
```

Comment:
I'd rather `GetTypeInfoByOid()` didn't return `dataTypeName`, which can simply be accessed on `postgresType`. At least in theory that method is generic, and the name is useful only in this specific context of parameter derivation. It would also simplify `GetTypeInfoByOid()` and make clear exactly what the `dataTypeName` is and where its coming from (i.e. that it's just a property on the `postgresType`.

Reply:
Agreed and fixed.